app/controllers/subscriptions_controller.rb

```diff
@@ -50,28 +50,25 @@ def update
   def unsubscribe
     subscription = Subscription.find_and_verify_by_token(token)
     @subscription = SubscriptionPresenter.new(subscription)
-
     Auditor::Audit.new(subscription, "subscription.#{subscription.frequency}_alert.delete", current_session_id).log
     subscription.delete
   end
 
 private
 
-  def token
-    ParameterSanitiser.call(params).require(:id)
+  def origin_param
+    params.permit(:origin)
   end
 
-  def subscription_params
-    ParameterSanitiser.call(params.require(:subscription)).permit(:email, :frequency, :search_criteria)
+  def search_criteria_params
+    ParameterSanitiser.call(params.require(:search_criteria).permit(:keyword, :location, :radius, job_roles: [], phases: [], working_patterns: []))
```

Comment:
am i right in thinking this was the part that was making those job_roles etc appear incorrectly on confirmation page?